File: src/main/java/ar/edu/um/tesoreria/rest/model/ChequeraEliminada.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@Entity
@Table(uniqueConstraints = {
		@UniqueConstraint(columnNames = { "facultadId", "tipoChequeraId", "chequeraSerieId" }) })
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
public class ChequeraEliminada extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 7355704417901165054L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long chequeraEliminadaId;

	private Integer facultadId;
	private Integer tipoChequeraId;
	private Long chequeraSerieId;
	private BigDecimal personaId;
	private Integer documentoId;
	private Integer lectivoId;
	private Integer arancelTipoId;
	private Integer cursoId;
	private Byte asentado;
	private Integer geograficaId;

	@JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
	private OffsetDateTime fecha;

	private Integer cuotasPagadas;
	private String observaciones;
	private Integer alternativaId;
	private Byte algoPagado;
	private Integer tipoImpresionId;
	private String usuarioId;

}

File: src/main/java/um/tesoreria/rest/kotlin/model/ChequeraEliminada.kt
package um.tesoreria.rest.kotlin.model

import um.tesoreria.rest.model.Auditable
import com.fasterxml.jackson.annotation.JsonFormat
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import java.math.BigDecimal
import java.time.OffsetDateTime

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["facultadId", "tipoChequeraId", "chequeraSerieId"])])
data class ChequeraEliminada(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var chequeraEliminadaId: Long? = null,

    var facultadId: Int? = null,
    var tipoChequeraId: Int? = null,
    var chequeraSerieId: Long? = null,
    var personaId: BigDecimal? = null,
    var documentoId: Int? = null,
    var lectivoId: Int? = null,
    var arancelTipoId: Int? = null,
    var cursoId: Int? = null,
    var asentado: Byte = 0,
    var geograficaId: Int? = null,

    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ssZ", timezone = "UTC")
    var fecha: OffsetDateTime? = null,

    var cuotasPagadas: Int = 0,
    var observaciones: String = "",
    var alternativaId: Int? = null,
    var algoPagado: Byte = 0,
    var tipoImpresionId: Int? = null,
    var usuarioId: String? = null,

    ) : Auditable()
